src/components/CookieConsent/ConsentManager.ts

```diff
@@ -0,0 +1,173 @@
+import pick from 'lodash/pick';
+import Cookies from 'universal-cookie';
+import type {CookieSetOptions} from 'universal-cookie/cjs/types';
+
+import type {IConsentManager, Subscriber} from './types';
+
+export const COOKIE_NAME = 'analyticsConsents';
+export const CONSENT_COOKIE_SETTINGS: CookieSettings = {
+    path: '/',
+    maxAge: 60 * 60 * 24 * 365,
+    secure: true,
+    sameSite: true,
+};
+
+export enum ConsentType {
+    Necessary = 'necessary',
+    Analytics = 'analytics',
+    Marketing = 'marketing',
+}
+
+export enum ConsentMode {
+    Notification = 'notification',
+    OptIn = 'opt-in',
+    Base = 'base',
+}
+
+export enum AdditionalConsentParams {
+    Closed = 'closed',
+    Edition = 'edition',
+}
+
+export type Consents = {
+    [k in `${ConsentType | AdditionalConsentParams}`]?: boolean | number;
+};
+
+export type CookieSettings = CookieSetOptions;
+
+const cookies = new Cookies();
+
+export class ConsentManager implements IConsentManager {
+    private consentMode: `${ConsentMode}`;
+    private consentEdition: number | undefined;
+    private projectConsentEdition: number | undefined;
+
+    private closed = false;
+    private consents: Consents = {};
+    private readonly cookiesTypes: Array<ConsentType> = Object.values(ConsentType);
+    private readonly subscribers: Subscriber[] = [];
+    private readonly cookieSettings: CookieSettings = CONSENT_COOKIE_SETTINGS;
+
+    constructor(mode: `${ConsentMode}`, edition?: number) {
+        this.consentMode = mode;
+        this.projectConsentEdition = edition;
+
+        this.setInitValues();
+    }
+
+    get mode() {
+        return this.consentMode;
+    }
+
+    set mode(newMode: `${ConsentMode}`) {
+        this.consentMode = newMode;
+    }
+
+    get cookies() {
+        return this.cookiesTypes;
+    }
+
+    get cookiesSettings() {
+        return this.cookieSettings;
+    }
+
+    getConsents() {
+        return {...this.consents};
+    }
+
+    subscribe(handler: Subscriber) {
+        this.subscribers.push(handler);
+
+        return () => {
+            const index = this.subscribers.findIndex((value) => value === handler);
+            if (index >= 0) {
+                this.subscribers.splice(index, 1);
+            }
+        };
+    }
+
+    setConsents(consents: Consents) {
```

Comment:
How about this:
```suggestion
    setConsents(consents: Consents | 'All' | 'OnlyNecessary') {
```

and make `prepareConsent` private?

Reply:
fixed